ArgumentValidator.php

```diff
@@ -153,4 +165,26 @@ private function validateClass($expectedClassName, $actualClassName)
             ));
         }
     }
+
+    /**
+     * Checks if the installed ExpressionLanguage component version is supported
+     * by the installed version of the DependencyInjection component.
+     *
+     * Starting with Symfony 2.6, the DependencyInjection component requires the
+     * ExpressionLanguage component to be of version 2.6 or higher.
+     *
+     * @return bool
+     */
+    private function isExpressionLanguageSupported()
+    {
+        if (!method_exists('Symfony\Component\DependencyInjection\Definition', 'getFactory')) {
```

Comment:
Aha, I understand it now! I think that incompatible dependencies is something that we shouldn't worry about at code level in this library. This is something which Symfony itself should prevent. So we should revert these code changes and just make sure that we don't install any incompatible dependencies while running the tests on Travis.